pkg/operator/hive/externaldns.go

```diff
@@ -54,7 +54,12 @@ func (r *ReconcileHiveConfig) deployExternalDNS(hLog log.FieldLogger, h *resourc
 	deployment := resourceread.ReadDeploymentV1OrDie(asset)
 
 	// Make AWS-specific changes to external-dns deployment
-	deployment.Spec.Template.Spec.Containers[0].Args = append(deployment.Spec.Template.Spec.Containers[0].Args, "--provider=aws")
+	args := deployment.Spec.Template.Spec.Containers[0].Args
+	args = append(args, "--provider=aws")
+	for _, domain := range instance.Spec.ManagedDomains {
+		args = append(args, fmt.Sprintf("--domain-filter=%s", domain))
```

Comment:
Verify that external-dns allows setting this parameter multiple times (that it's not "last one wins").

Reply:
I verified that this behaves as expected. Multiple arguments result in a union.